js/search.js

```diff
@@ -962,37 +997,46 @@
 				if (id === 'icywind') isViable = (toId(set.species).substr(0, 6) === 'keldeo');
+				if (id === 'inferno') isViable = (format === 'mixandmega' && toId(set.item) === 'pidgeotite' && moves.indexOf('fireblast') < 0);
 				if (id === 'infestation') isViable = (toId(set.species) === 'shuckle');
 				if (id === 'irontail') isViable = ((template.types.indexOf('steel') > 0 && moves.indexOf('ironhead') < 0) || ((template.types.indexOf('dark') > 0 || template.types.indexOf('dragon') > 0) && moves.indexOf('ironhead') < 0 && moves.indexOf('gunkshot') < 0));
 				if (id === 'jumpkick') isViable = (moves.indexOf('highjumpkick') < 0);
 				if (id === 'leechlife') isViable = (this.gen > 6);
 				if (id === 'petaldance') isViable = (toId(set.ability) === 'owntempo');
 				if (id === 'reflecttype') isViable = (toId(set.species) in {latias:1, starmie:1});
-				if (id === 'rocktomb') isViable = (toId(set.species) === 'groudon');
+				if (id === 'rocktomb') isViable = (toId(set.species) === 'groudon' || toId(set.ability) === 'technician');
 				if (id === 'selfdestruct') isViable = (this.gen < 5 && moves.indexOf('explosion') < 0);
+				if (id === 'sing') isViable = (format === 'mixandmega' && toId(set.item) === 'pidgeotite');
 				if (id === 'skyattack') isViable = (toId(set.species) === 'hawlucha');
 				if (id === 'smackdown') isViable = (template.types.indexOf('ground') > 0);
 				if (id === 'smartstrike') isViable = (template.types.indexOf('steel') > 0 && moves.indexOf('ironhead') < 0);
 				if (id === 'solarbeam') isViable = (toId(set.abilities) in {drought:1, chlorophyll:1});
+				if (id === 'stompingtantrum') isViable = ((moves.indexOf('earthquake') < 0 && moves.indexOf('drillrun') < 0) || (toId(set.ability) === 'toughclaws' && moves.indexOf('drillrun') < 0 && moves.indexOf('earthquake') < 0));
 				if (id === 'storedpower') isViable = (toId(set.species) in {necrozma:1, espeon:1, sigilyph:1});
 				if (id === 'stunspore') isViable = (moves.indexOf('thunderwave') < 0);
-				if (id === 'thunder') isViable = (toId(set.ability) === 'drizzle' || (toId(set.ability) === 'primordialsea') || (toId(set.species) === 'xerneas'));
+				if (id === 'thunder') isViable = (toId(set.ability) === 'drizzle' || (toId(set.ability) === 'primordialsea') || (toId(set.species) === 'xerneas') || (format === 'mixandmega' && toId(set.item) === 'pidgeotite' && moves.indexOf('zapcannon') < 0));
 				if (id === 'trickroom') isViable = (template.baseStats.spe <= 100);
+				if (id === 'waterpulse') isViable = ((toId(set.ability) === 'megalauncher' && moves.indexOf('primordialsea') < 0) || (format === 'mixandmega' && toId(set.item) === 'blastoisinite' && moves.indexOf('primordialsea') < 0));
+				if (id === 'weatherball') isViable = (format === 'mixandmega' && toId(set.item) === 'redorb');
+				if (id === 'zapcannon') isViable = (format === 'mixandmega' && toId(set.item) === 'pidgeotite');
```

Comment:
I would put the M&M viabilities in an `if (format === 'mixandmega') {` block, instead of interspersing them.

Reply:
This causes a crash

Reply:
I’ll update the commit after school

Reply:
What's the crash message?